Update Test Verification section (#3227)
magento/pwa-studio: .github/PULL_REQUEST_TEMPLATE.md

<!-- Changes to `upward-spec` and/or `upward-js` packages must be reviewed -->
<!-- by `UPWARD-PHP` maintainers to ensure continued compatibility -->

## Test Plan

> <!-- We look forward to have no/less bugs part of our code review and QA -->

Reviewer: ```suggestion
<!-- During code review and QA we will try to ensure there are no bugs introduced by this change -->
```

> <!-- So, we request you to add detailed test plan on what needs to be QA'ed before this PR gets merged -->

Reviewer: ```suggestion
<!-- So, we request that you add a detailed test plan on what needs to be checked before this PR gets merged -->
```

> <!-- As you add the test plan you would also come across a scenario which might needs to checked -->
> <!-- Also, as part of Review/QA we will update test plan if needed -->

Reviewer: ```suggestion
<!-- Feel free to update this after submitting the PR as you discover new scenarios -->
<!-- As part of review/QA we may also add or update the test plan if necessary-->
```

> #### Test scenario(s) for direct fix/feature - 

Reviewer: ```suggestion
#### Test scenario(s) for direct fix/feature
```

Example
-   Verify user is able to apply filters on category page
-   Verify user is able to apply filters on search page

> #### Test scenario(s) for any existing impacted features/areas - 

Reviewer: ```suggestion
#### Test scenario(s) for any existing impacted features/areas
```

Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

> #### Test scenario(s) for any Magento Backend Supported Configurations -

Reviewer: ```suggestion
#### Test scenario(s) for any Magento Backend Supported Configurations
```

Example
- Update default Sort value in backend and repeat above scenarios

> #### Does Browser/Device testing needed?

Reviewer: ```suggestion
#### Is Browser/Device testing needed?
```

Example

> - Yes, Browser testing is needed as X UI component is impacted 

Reviewer: ```suggestion
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>
```

> #### Any adhoc/edge case scenarios needs to be considered?

Reviewer: ```suggestion
#### Any ad-hoc/edge case scenarios that need to be considered?
```
